dev_tools/conftest.py

```diff
@@ -29,3 +39,85 @@ def pytest_collection_modifyitems(config, items):
     for item in items:
         if 'slow' in item.keywords:
             item.add_marker(skip_slow_marker)
+
+
+@pytest.fixture(scope="session")
+def cloned_env(testrun_uid, worker_id):
+    """Fixture to allow tests to run in a clean virtual env.
+
+    It de-duplicates installation of base packages. Assuming `virtualenv-clone` exists on the PATH,
+    it creates first a prototype environment and then clones for each new request the same env.
+    This fixture is safe to use with parallel execution, i.e. pytest-xdist. The workers synchronize
+    via a file lock, the first worker will (re)create the prototype environment, the others will
+    reuse it via cloning.
+
+    A group of tests that share the same base environment is identified by a name, `env_dir`,
+    which will become the directory within the temporary directory to hold the virtualenv.
+
+    Usage:
+
+    >>> def test_something_in_clean_env(cloned_env):
+            # base_env will point to a pathlib.Path containing the virtual env which will
+            # have quimb, jinja and whatever reqs.txt contained installed.
+            base_env = cloned_env(env_dir = "some_tests", "quimb", "jinja", "-r", "reqs.txt")
+
+            # To install new packages (that are potentially different for each test instance)
+            # just run pip install from the virtual env
+            subprocess.run(f'{base_env}/bin/pip install something', shell=True)
```

Comment:
if you read the docs, `shell=True` is discouraged 1) do we need shell=True or are you just getting around doing "command arg arg".split() 2) if it's required, please add a note why

Reply:
+1 - thanks for the pointer - didn't know. Got rid of it!